### ISSUE TITLE
Variables are not highlighted in redirections

### ISSUE THREAD

**Issue by zx8** [fish-shell/fish-shell#8864]
```
$ fish --version
fish, version 3.3.1-1246-g380c55584
```

---

<img width="222" alt="image" src="https://user-images.githubusercontent.com/8544900/162236142-5f6fd899-8edd-41a2-b897-c4989ec0b561.png">

I would expect the `$bar` in `$bar.yaml` to be highlighted in yellow.

Is this a regression, or has it always worked like this?

If it has always worked like this, please consider this a feature request rather than a bug report. 😄

**Comment by faho** [comment]
Neither - since we can only highlight for one thing, we highlight this as a valid or invalid redirection - if the target ends up empty (because the variable isn't defined), it's invalid and highlighted with $fish_color_error. If the variable is defined, or will be defined by the time the redirection happens (#6654), we highlight it as a valid redirection with $fish_color_redirection.

I feel that's the correct idea - redirections > variables, because the redirection is about the context of the entire token. I don't believe adding another color $fish_color_variable_in_redirection would be a good idea.

So I'll be closing this.

**Comment by Nahor** [comment]
I don't think it needs a new color, it can use the same color as the other variables.

>because the redirection is about the context of the entire token

Isn't it similar for commands? Aren't commands "tokens"? If the command doesn't exist, the whole command is red. But if the command is valid and contains a variable, the variable has it's own color instead of using the command color.

<img width="499" height="132" alt="Image" src="https://github.com/user-attachments/assets/694a8009-e62a-46ac-a862-6222a9b4dd61" />

It feels to me that redirection ought to be doing the same thing.

Another reason is that variables have "priority" over the redirection, they are expanded before the redirection is applied. So it would make sense for the variable color to also apply with priority over the redirection color.


And for that matter, I believe it should also apply the quoted string coloring (although I only feel mildly about that one).

**Comment by krobelus** [comment]
yeah it's weird that redirection and command coloring is inconsistent
